step-capstone/src/components/Utilities/PreferenceForm.js

```diff
@@ -0,0 +1,260 @@
+    FormControlLabel,
+    FormHelperText,
+    MenuItem,
+    InputLabel,
+    Select,
+    Slider,
+    FormLabel,
+    FormGroup,
+    Checkbox
+} from "@material-ui/core"
+
+export default class PreferenceForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            budget: this.props.pref.budget,
+            radius: this.props.pref.radius,
+            activityPreferences: this.props.pref.activityPreferences,
+            foodPreferences: this.props.pref.foodPreferences
+        }
+    }
+
+    componentDidUpdate(prevState) {
+        if (prevState !== this.state) {
+            this.props.onChange(this.state);
+        }
+    }
+
+    handleBudgetChange = (event) => {
+        this.setState({ budget: event.target.value });
+    }
+
+    handleRadiusChange = (event, newValue) => {
+        this.setState({ radius: newValue })
+    }
+    ß
+    handleActivityPreferenceCheck = (event) => {
+        this.togglePreferenceHelper("activityPreferences", event.target.name)
+    }
+
+    handleFoodPreferenceCheck = (event) => {
+        this.togglePreferenceHelper("foodPreferences", event.target.name)
+    }
+
+    togglePreferenceHelper = (listName, pref) => {
+        if (!this.state[listName].includes(pref)) {
+            this.setState({ [listName]: this.state[listName].concat([pref]) })
+        } else {
+            this.setState({ [listName]: this.state[listName].filter(activity => activity !== pref) })
+        }
+    }
+
+    render() {
+        return (
+            <Box mt={2}>
+                <Grid container>
+                    <Grid item>
+                        <Typography variant="h5" gutterBottom>Preferences</Typography>
+                        <Typography variant="subtitle2" gutterBottom>Select your own preferences and we'll give you custom suggestions for foor and activities!</Typography>
+                    </Grid>
+                    <Grid container item direction="column">
+                        <Grid item>
+                            <Box mb={1}>
+                                <FormControl>
+                                    <InputLabel>Budget</InputLabel>
+                                    <Select
+                                        value={this.state.budget}
+                                        onChange={this.handleBudgetChange}
+                                        displayEmpty
+                                    >
+                                        <MenuItem value={0}></MenuItem>
+                                        <MenuItem value={1}>$</MenuItem>
+                                        <MenuItem value={2}>$$</MenuItem>
+                                        <MenuItem value={3}>$$$</MenuItem>
+                                        <MenuItem value={4}>$$$$</MenuItem>
+                                    </Select>
+                                    <FormHelperText>We won't give you anything above this price!</FormHelperText>
```

Comment:
technically we will give them things above the price right? just later in the suggestions? maybe we can change the message? otherwise, the budget is probably fairly self explanatory.

Reply:
True! Probably don't need helper text